orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileCacheManagerImpl.java

```diff
@@ -155,11 +171,10 @@ public void put(OrcidProfile orcidProfile) {
 
     public void put(String orcid, OrcidProfile orcidProfile) {
         try {
-            synchronized (profileLockers.obtainLock(orcid)) {
-                profileCache.put(new Element(new OrcidCacheKey(orcid, releaseName), orcidProfile));
-            }
+            profileCache.acquireWriteLockOnKey(orcid);
```

Comment:
Hey @rcpeters, i think here you should create the key _new OrcidCacheKey(orcid, releaseName)_ and acquire the lock on that key, not in the orcid id right?